src/components/CaptureMatching/CaptureMatchingView.js

```diff
@@ -673,14 +739,14 @@ function CaptureMatchingView() {
             {loading ? null : (
               <CandidateImages
                 capture={captureImage}
-                candidateImgData={candidateImgData}
+                candidateImgData={filterTree(candidateImgData, captureImage)}
```

Comment:
We're performs the filtering multiple times on every render – that seems inefficient.
Instead, if we still need it, it should be performed in a hook when `candidateImgData` changes.